### PR TITLE
Add support for live-reloading of modified HTML/CSS files for uSciter.

### PR DESCRIPTION
**Pull request by pravic**
Andrew, a feature to consider.

This PR adds a new checkbox for uSciter which enables live-reloading of the current file.

The main purpose is to reload the current file as soon as it or any dependent HTML or CSS files are  modified. 

I intentionally skip TIS files because they (usually) do not affect UI but they are modified quite frequently. However, may be it is reasonable to monitor graphics resources (PNG/SVG) as well.

Essentially live-reloading works as this:

 * it installs a new `view.onRequestResponse` hook in order to track the all requests to local (e.g. file://) HTML and CSS files that are loaded in uSciter's view (of course, the old hook is preserved because it is needed by Inspector)
 * each tracked file is checked every second whether is has been modified
 * if has, a delayed reload of the main file is scheduled

This feature request is opened for discussion and the code can be changed. Also I realize that uSciter is quite simple and may be there is no intention to evolve it.

### CODE REVIEW

**Comment by c-smile** [comment]
Why not to use FileMonitor :  `System.watch(path)` ?
See:
https://sciter.com/docs/content/script/System.htm

**Comment by pravic** [comment]
It's easier to check a few needed files rather than to monitor all the modified files in the directory.

Also, what exactly directory to register for System.watch? The most common ancestor for all tracked files? Because they can lay in various directories.

**Comment by c-smile** [comment]
FileMonitor when it is set on a directory tracks any changes of any file in that directory.
Monitor is shallow - it does not go inside sub-directories. 

Consider /images/ folder for example. You may have written something like   that uses foo.png that is not there yet. Ideally the html shall be refreshed to show the image when it will put there, no?

**Comment by pravic** [comment]
So, the file monitor doesn't track subdirectories - it have to be set for each parent directory of each tracked resource. 

However, the image example makes sense. I should add the support of new files. I'll rewrite the code using the file watch, thanks!

File: demos/usciter/res/live-reload.tis [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    this.urls[url] = path;
+    debug(live: add {url} as {path});
+
+    if(!this.watchers[path]) {
+      this.watchers[path] =
+
+        System.watch(path) << event change (filename) {
+          var path = this.path + "/" + filename;
+          debug(live: change in {path});
```
Comment (pravic) [inline demos/usciter/res/live-reload.tis:64]: @c-smile Andrew, is there any reason that `debug` doesn't work in this place? I've tried to add different debug statements, neither of them work. That's why `stdout` is there.

File: demos/usciter/res/live-reload.tis [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        if(rt == Request.DATA_HTML || rt == Request.DATA_STYLE || rt == Request.DATA_IMAGE)
+          return true;
+
+        return false;
+      }
+
+      // check this resource
+      var url = rq.responseUrl || rq.requestUrl;
+      var reloadable = isTracked(rq, url);
+      // debug({url} rq {rq.status} {rq.isConsumed});
```
Comment (pravic) [inline demos/usciter/res/live-reload.tis:115]: Why is  `onRequestResponse` called twice, by the way? The last one is with the consumed bit set.

File: demos/usciter/res/live-reload.tis (hunk 1 above)
Comment (c-smile) [reply demos/usciter/res/live-reload.tis:64]: Not sure, the only thing: debug message queue is associated with debugee document. When you reload document the queue is dropped.

Comment (pravic) [reply demos/usciter/res/live-reload.tis:64]: No, if I comment the reloading, it still doesn't print debug messages..

Comment (c-smile) [reply demos/usciter/res/live-reload.tis:64]: The live-reload feature landed in r7741 + associated fixes.

File: demos/usciter/res/live-reload.tis (hunk 2 above)
Comment (c-smile) [reply demos/usciter/res/live-reload.tis:115]: Also fixed by r7741

File: demos/usciter/res/live-reload.tis (hunk 1 above)
Comment (pravic) [reply demos/usciter/res/live-reload.tis:64]: I've checked it. Looks great. Thanks!

**Comment by pravic** [comment]
Well, closing this in favor of upstream changes :+1:

**Pull request closed without merging**